src/main/java/mainstructure/Geo.java

```diff
@@ -0,0 +1,148 @@
+                        UI.print("""
+                        You need to specify the deadline!
+                        Format: deadline [task name] /by [deadline]
+                        """);
+                        break;
+                    case "event":
+                        UI.print("""
+                        You need to specify the start and end date of the event!
+                        Format: event [task name] /from [start time] /to [end time]
+                        """);
+                        break;
+                    case "delete":
+                        UI.print("""
+                        You need to specify which task to delete!
+                        Format: delete X
+                        """);
+                        break;
+                    default:
+                        //Not expecting to execute this
+                        UI.print("Seems like an error occurred in the code");
+                        break;
+                    }
+                } catch (IOException e){
+                    UI.print("Saving error");
+                }
+            }
+             */
+        }
+    }
+
+    private static void markTask(TaskList list, String input, boolean status){
+        try {
+            int index = Integer.parseInt(input.substring(status? 5 : 7));
+            list.mark(index, status);
+        } catch (NumberFormatException | StringIndexOutOfBoundsException e){
+            //If input missing information
+            UI.print("""
+                    You need to specify which task to be marked!
+                    Format: mark/unmark X
+                    """);
+        } catch (NullPointerException e){
+            //Not expecting to execute this because the mark() handles errors already
+            UI.print("Error occurred in taskList\n");
+        }
+    }
+
+    public static void addTask(TaskList list, String command, String rest) throws IOException {
+        try {
+            switch (command) {
+            case "todo":
+                list.addTask(rest);
+                SaveManager.appendTextFile(SaveManager.defaultPath, list.showTask(list.getTaskCount()));
+                break;
+            case "deadline":
+                String[] split = rest.split("/by|\\(by:|\\)", 3);
+                list.addTask(split[0].trim(), split[1].trim());
+                SaveManager.appendTextFile(SaveManager.defaultPath, list.showTask(list.getTaskCount()));
+                break;
+            case "event":
+                String[] firstSplit = rest.split("/from|\\(from:", 2);
+                String[] secondSplit = firstSplit[1].split("/to|to:|\\)", 3);
+                list.addTask(firstSplit[0].trim(), secondSplit[0].trim(), secondSplit[1].trim());
+                SaveManager.appendTextFile(SaveManager.defaultPath, list.showTask(list.getTaskCount()));
+                break;
+            default:
+                //Not expecting to execute this
```

Comment:
The comments should be for the reader. This comments seems to be for your own reference.